src/objects.rs

```diff
@@ -295,43 +554,60 @@ pub enum CompareType {
     Or,
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, Copy, Deserialize, Serialize)]
-pub struct CircuitCondition;
-
 #[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize)]
+pub struct CircuitCondition {
+    pub comparator: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub constant: Option<i32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub first_signal: Option<serde_json::value::Value>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub second_signal: Option<serde_json::value::Value>,
```

Comment:
Do we have examples which show the JSON that these values correspond to? Leaving them as `Value`s seems suboptimal.

Reply:
Whoops, these are just `SimpleEntity`s